src/new-client/src/app/lib/services/viewAvailabilityService.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  AvailabilityBlock,
+  AvailabilityResponse,
+  Booking,
+  clinicalServices,
+  FetchBookingsRequest,
+  Week,
+} from '@types';
+import dayjs from 'dayjs';
+import { fetchBookings } from './appointmentsService';
+
+const BuildWeeks = (year: number, month: number): Week[] => {
+  const weeks: Week[] = [];
+
+  const [lastDateMonth, lastDateYear] =
+    month === 11 ? [0, year + 1] : [month + 1, year];
+  const firstDate = new Date(year, month, 1);
+  const lastDate = new Date(lastDateYear, lastDateMonth, 0);
+  const numDays = lastDate.getDate();
+  const dayOfWeekCounter = firstDate.getDay();
+
+  let start = 1;
+  let end = 7;
+  if (dayOfWeekCounter === 0) {
+    end = 1;
+  } else {
+    end = 7 - dayOfWeekCounter + 1;
+  }
+
+  while (start <= numDays) {
+    weeks.push({
+      start: start,
+      end: end,
+      startMonth: month,
+      endMonth: month,
+      startYear: year,
+      endYear: lastDateYear,
+      bookedAppointments: [],
+    });
+    start = end + 1;
+    end = end + 7;
+    end = start === 1 && end === 8 ? 1 : end;
+    if (end > numDays) {
+      end = numDays;
+    }
+  }
+
+  return weeks;
+};
+
+export const getWeeksInMonth = (year: number, month: number): Week[] => {
```

Comment:
We're fairly committed to using DayJS as our datetime handler elsewhere in the code, so it might be nice to be consistent here rather than using JS' own date object. 

Please ignore this if not helpful, but you might actually be able to adapt some similar methods that were also written for a [previous prototype](src/new-client/src/app/lib/services/timeService.ts), e.g. 
<img width="412" alt="image" src="https://github.com/user-attachments/assets/3353096f-8057-46c1-835a-4103eec1d515">